draft-ietf-jsonpath-base.md

```diff
@@ -1502,10 +1502,24 @@ The well-typedness of function expressions can now be defined in terms of this t
 
 For a function expression to be well typed:
 
-1. its declared type must be well typed in the context in which it occurs, and
-2. its arguments must be well typed for the declared type of the corresponding parameters.
+1. its function name MUST be registered (see {{iana-fnex}}),
+2. its declared type must be well typed in the context in which it occurs, and
+3. its arguments must be well typed for the declared type of the corresponding parameters.
 
-(1) As per the grammar, a function expression can occur in three different
+(1) The registered set of function extensions is the source of knowledge regarding the declared return and
+parameter types of a function extension.
+
+If a function expression is treated as well typed, the function extensions involved in the expression belong
+to the registered set.
```

Comment:
The way this logic is stated means that well-typedness automatically implies membership in the registry.  That's not precisely the case.

I think what you're meaning is that membership in the registry is a prerequisite to being considered well-typed.

Reply:
> The way this logic is stated means that well-typedness automatically implies membership in the registry. That's not precisely the case.

Notice the phrase "treated as well typed". What's incorrect about the statement?

> 
> I think what you're meaning is that membership in the registry is a prerequisite to being considered well-typed.

That's certainly true, but I also had in mind the effect on the user. If a query succeeds, the user can safely assume that any function extensions in the query are registered. Perhaps I need to spell that out.